enumerable.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each(array)
+    array.length.times { |index| yield(array[index]) }
+
+    nil
+  end
+
+  def my_each_with_index(array)
+    0.upto(array.length - 1) { |index| yield(array[index], index) }
+
+    nil
+  end
+
+  def my_select(array)
+    result = []
+
+    my_each(array) { |item| result << item if yield(item) }
+
+    result
+  end
+
+  def my_all?(array)
+    result = true
+
+    my_each(array) { |item| result = false unless yield(item) }
+
+    result
+  end
+
+  def my_any?(array)
+    result = false
+
+    my_each(array) { |item| result = true if yield(item) }
+
+    result
+  end
+
+  def my_none?(array)
+    result = true
+
+    my_each(array) { |item| result = false if yield(item) }
+
+    result
+  end
+
+  def my_count(array, value = nil)
+    counter = 0
+
+    my_each(array) do |item|
+      if block_given?
+        counter += 1 if yield(item)
+      elsif item == value
+        counter += 1
+      else
+        counter = array.length unless value
+      end
+    end
+    counter
+  end
+
+  def my_map(array)
+    result = []
+
+    my_each(array) { |item| result << yield(item) }
+
+    result
+  end
+
+  def my_inject(array, value = nil)
+    if value
+      result = value
+
+      my_each(array) { |item| result = yield(result, item) }
+    else
+      result = array[0]
+      array.shift
+
+      my_each_with_index(array) { |item, _index| result = yield(result, item) }
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } ==l (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77698374-1eb19500-6f76-11ea-8b2c-5d532302ce65.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77698501-56b8d800-6f76-11ea-937a-975824cfa279.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)